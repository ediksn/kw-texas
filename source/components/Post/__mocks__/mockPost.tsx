const mockPost = {
  id: 1,
  author: {
    name: 'Nina Royals',
    picture: '__mocks__/avatarNinaRoyals.png'
  },
  content:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean tellus leo, fringilla at elit sed, tincidunt viverra dui. Nullam dignissim mi et tincidunt luctus. Nullam ut eleifend diam. Phasellus faucibus aliquam leo, vitae scelerisque nunc dignissim in. Maecenas vulputate quam id metus efficitur tristique. Ut maximus varius enim eleifend feugiat. Ut at ultricies lorem, eget molestie risus. Nam ultricies tempus sem quis fermentum. Sed a eros ac metus scelerisque hendrerit. Ut nec urna aliquam, sagittis eros non, bibendum risus. Ut congue eleifend velit eu dictum. Praesent eu mi vitae leo pulvinar ullamcorper. Sed tortor felis, tristique vitae lacinia sit amet, pellentesque ut felis. Nulla finibus erat libero, eu sodales metus auctor at. Duis iaculis leo vitae odio mollis, feugiat aliquam erat eleifend. Mauris quis dui at lorem tincidunt viverra nec vel mauris. Pellentesque laoreet dapibus purus, eget vehicula nunc scelerisque quis. Donec ac nibh sed erat consequat sagittis ut ut odio. Curabitur tempor vitae neque vel gravida. Morbi varius ligula nec tellus accumsan, quis pharetra ante dictum. Nulla pharetra ac ex id cursus. Vivamus ut lacinia nunc. Maecenas nec orci risus.',
  date: '4 hours ago',
  likes: 13,
  comments: 5,
  shares: 2
}

export default mockPost
