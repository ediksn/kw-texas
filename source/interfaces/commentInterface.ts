export interface CommentDetailInterface {
  content: string
  edited: boolean
  likesCount: number
  userHasAlreadyLiked: boolean
  attachment: any
}

export interface CommentInterface {
  id: string
  creatorId: string
  createdAt: string
  type: string
  creatorPhoto: string
  creatorFirstName: string
  creatorLastName: string
  deleted: boolean
  postId: string
  updatedAt: string
  detail: CommentDetailInterface
}
