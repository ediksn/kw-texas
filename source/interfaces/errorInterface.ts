export interface ErrorInterface {
  error_description: string | null
  error: string | null
}
