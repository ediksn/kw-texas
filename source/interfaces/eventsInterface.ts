import moment from 'moment'

interface InstructorsInterface {
  firstName: string
  lastName: string
}

export interface EventsInterface {
  id: number
  name: string
  price: string
  location: string
  ends: moment.Moment
  starts: moment.Moment
  published: boolean
  instructors: InstructorsInterface[]
  bookmarked: boolean
  slug: any
}
