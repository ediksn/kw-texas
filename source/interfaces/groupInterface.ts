import { TextStyle, ViewStyle } from 'react-native'

export interface OptionInterface {
  key?: string
  isTitle?: boolean
  handleOption?: () => void
  title?: string
  color?: string
}

export interface GroupOptionInterface {
  visible: boolean
  buttonRef: any
  options: OptionInterface[]
  styleContainer?: ViewStyle
  styleOption?: TextStyle
  toRight?: boolean
  onRequestClose: () => void
}

interface Icon {
  url: string
  id: string
  filename: string
}

interface LocationInterface {
  city: string
  state: string
  postalCode: string
  country: string
}

interface MemberInterface {
  id: string
  kwUid: number
  username: string
  firstName: string
  lastName: string
  email: string
  memberType: string
  state: string
  phone: string
  photoUrl: string
  location: LocationInterface
}
export interface GroupInterface {
  id: string
  name: string
  description: string
  postCount: number
  membersCount: number
  status: number
  icon: Icon
  members: MemberInterface[]
}

export interface GroupProduceProps {
  type: string
  payload?: {
    data: GroupInterface[]
    limit: number
  }
}

export interface GroupReducerProps {
  draftState: any
  payload?: {
    data: GroupInterface[]
    limit: number
  }
}
