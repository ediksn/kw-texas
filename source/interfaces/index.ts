export { default as loginInterface } from './loginInterface'
export { default as promptVideoInterface } from './promptVideoInterface'
