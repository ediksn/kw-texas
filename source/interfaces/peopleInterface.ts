export interface PeopleMarkedInterface {
  [kwUid: number]: boolean
}

interface LocationInterface {
  country: string | null
  state: string
  city: string
  address: string | null
  postalCode: string | null
}
export interface PeopleInterface {
  kwUid: number
  firstName: string
  lastName: string
  location: LocationInterface
  photoUrl: string
}

export interface PeopleProduceProps {
  type: string
  payload?: {
    data: PeopleInterface[]
    limit: number
  }
}

export interface PeopleReducerProps {
  draftState: any
  payload?: any
}
