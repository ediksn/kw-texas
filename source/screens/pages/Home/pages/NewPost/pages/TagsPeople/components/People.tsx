import React from 'react'
import { TouchableOpacity, Text } from 'react-native'
import { CheckBox, Avatar } from '~/components'
import { styles } from './styles'
import { PeopleInterface } from '~/interfaces/peopleInterface'

interface PeopleProps {
  people: PeopleInterface
  handlePeopleMarked: (kwUid: number) => void
  marked?: boolean
}

const People = ({ people, handlePeopleMarked, marked }: PeopleProps) => {
  const { kwUid, photoUrl, firstName, lastName } = people

  return (
    <TouchableOpacity onPress={() => handlePeopleMarked(kwUid)} style={styles.container}>
      <CheckBox marked={marked} />
      <Avatar
        uri={photoUrl}
        avatarStyle={styles.avatar}
        avatarDefaultStyle={styles.avatar}
        initialsStyle={styles.avatar}
      />
      <Text style={styles.name}>{`${firstName} ${lastName}`}</Text>
    </TouchableOpacity>
  )
}

export default People
