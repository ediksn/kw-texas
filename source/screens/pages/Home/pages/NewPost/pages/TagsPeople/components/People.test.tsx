import React from 'react'
import { render, RenderAPI } from '@testing-library/react-native'
import People from './People'
import { PeopleInterface } from '~/interfaces/peopleInterface'
import { peopleResponse } from '~/../__mocks__/mockResponses'

jest.mock('react-redux', () => ({
  useSelector: () => ({})
}))

describe('Render People Component Test', () => {
  let component: RenderAPI
  const mockPeople: PeopleInterface = peopleResponse.data.data.getAllUsers[0]

  beforeEach(() => {
    component = render(<People people={mockPeople} handlePeopleMarked={() => null} />)
  })

  it('Match with the snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('Renders correctly', () => {
    expect(component).toBeDefined()
  })
})
