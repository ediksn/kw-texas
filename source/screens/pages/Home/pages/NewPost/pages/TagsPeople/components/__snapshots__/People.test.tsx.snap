// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render People Component Test Match with the snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#E3E3E4",
      "borderStyle": "solid",
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "height": 82.39411764705882,
      "opacity": 1,
      "paddingLeft": 40.714285714285715,
      "paddingRight": 34.285714285714285,
    }
  }
>
  <Image
    source={
      Object {
        "process": [Function],
      }
    }
    style={
      Object {
        "height": 28.285714285714285,
        "width": 28.285714285714285,
      }
    }
  />
  <Image
    onError={[Function]}
    resizeMode="cover"
    source={
      Object {
        "uri": "https://avatar.kwconnect.com/6.jpeg",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 31.428571428571427,
          "height": 62.857142857142854,
          "width": 62.857142857142854,
        },
        Object {
          "borderRadius": 31.428571428571427,
          "height": 37.714285714285715,
          "marginLeft": 23.57142857142857,
          "marginRight": 17.142857142857142,
          "width": 37.714285714285715,
        },
      ]
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "Mulish-Bold",
        "fontSize": 22,
        "lineHeight": 20,
      }
    }
  >
    6 QA
  </Text>
</View>
`;
